### ISSUE TITLE
Cannot link filter to question on dashboard

### ISSUE THREAD

**Issue by iaindillingham** [metabase/metabase#13958]
**Describe the bug**
I cannot link a filter to a question on a dashboard. After creating the question, dashboard, and filter, clicking *Edit dashboard* > *Edit* (on the filter) does not replace the question with a drop-down, indicating the columns to which the filter could be applied (see attached).

**Logs**
Server logs for the sequence described above:
```
[c0da042c-4037-4d6f-be2f-b07bb24af21b] 2020-12-04T11:48:56+00:00 DEBUG metabase.middleware.log POST /api/card/126/query 202 [ASYNC: completed] 5.4 s (14 DB calls) App DB connections: 1/15 Jetty threads: 3/50 (17 idle, 0 queued) (122 total active threads) Queries in flight: 0 (0 queued)
```

JavaScript logs for the sequence described above:
```
DEPRECATED: metabase/redux/metadata fetchField app-main.bundle.js:5:101683
DEPRECATED: metabase/redux/metadata fetchFieldValues app-main.bundle.js:5:101683
Uncaught TypeError: this._query.table() is null
    value https://metabase.prod2.prod-penguin.co.uk/app/dist/app-main.bundle.js?31778b9e3b01ff7193ac:5
    value https://metabase.prod2.prod-penguin.co.uk/app/dist/app-main.bundle.js?31778b9e3b01ff7193ac:5
    v https://metabase.prod2.prod-penguin.co.uk/app/dist/app-main.bundle.js?31778b9e3b01ff7193ac:5
    value https://metabase.prod2.prod-penguin.co.uk/app/dist/app-main.bundle.js?31778b9e3b01ff7193ac:5
    getParameterMappingOptions https://metabase.prod2.prod-penguin.co.uk/app/dist/app-main.bundle.js?31778b9e3b01ff7193ac:5
    makeGetParameterMappingOptions https://metabase.prod2.prod-penguin.co.uk/app/dist/app-main.bundle.js?31778b9e3b01ff7193ac:5
    u https://metabase.prod2.prod-penguin.co.uk/app/dist/vendor.bundle.js?31778b9e3b01ff7193ac:1
    i https://metabase.prod2.prod-penguin.co.uk/app/dist/vendor.bundle.js?31778b9e3b01ff7193ac:1
app-main.bundle.js:5:139980
Missing id: 
Object { values: values() }
app-main.bundle.js:5:122968
Uncaught TypeError: null has no properties
    getHostNode https://metabase.prod2.prod-penguin.co.uk/app/dist/vendor.bundle.js?31778b9e3b01ff7193ac:1
    getHostNode https://metabase.prod2.prod-penguin.co.uk/app/dist/vendor.bundle.js?31778b9e3b01ff7193ac:1
    getHostNode https://metabase.prod2.prod-penguin.co.uk/app/dist/vendor.bundle.js?31778b9e3b01ff7193ac:1
    updateChildren https://metabase.prod2.prod-penguin.co.uk/app/dist/vendor.bundle.js?31778b9e3b01ff7193ac:1
    _reconcilerUpdateChildren https://metabase.prod2.prod-penguin.co.uk/app/dist/vendor.bundle.js?31778b9e3b01ff7193ac:1
    _updateChildren https://metabase.prod2.prod-penguin.co.uk/app/dist/vendor.bundle.js?31778b9e3b01ff7193ac:1
    updateChildren https://metabase.prod2.prod-penguin.co.uk/app/dist/vendor.bundle.js?31778b9e3b01ff7193ac:1
    _updateDOMChildren https://metabase.prod2.prod-penguin.co.uk/app/dist/vendor.bundle.js?31778b9e3b01ff7193ac:1
vendor.bundle.js:1:722648
Uncaught (in promise) TypeError: null has no properties
    getHostNode https://metabase.prod2.prod-penguin.co.uk/app/dist/vendor.bundle.js?31778b9e3b01ff7193ac:1
    getHostNode https://metabase.prod2.prod-penguin.co.uk/app/dist/vendor.bundle.js?31778b9e3b01ff7193ac:1
    getHostNode https://metabase.prod2.prod-penguin.co.uk/app/dist/vendor.bundle.js?31778b9e3b01ff7193ac:1
    updateChildren https://metabase.prod2.prod-penguin.co.uk/app/dist/vendor.bundle.js?31778b9e3b01ff7193ac:1
    _reconcilerUpdateChildren https://metabase.prod2.prod-penguin.co.uk/app/dist/vendor.bundle.js?31778b9e3b01ff7193ac:1
    _updateChildren https://metabase.prod2.prod-penguin.co.uk/app/dist/vendor.bundle.js?31778b9e3b01ff7193ac:1
    updateChildren https://metabase.prod2.prod-penguin.co.uk/app/dist/vendor.bundle.js?31778b9e3b01ff7193ac:1
    _updateDOMChildren https://metabase.prod2.prod-penguin.co.uk/app/dist/vendor.bundle.js?31778b9e3b01ff7193ac:1
vendor.bundle.js:1:722648
Security Error: Content at https://metabase.prod2.prod-penguin.co.uk/dashboard/31 may not load data from blob:moz-extension://98d4eefc-f14f-4405-828e-a098dd980f28/4393ebdd-6f31-40fd-bb84-44481549f5d0.
```

**To Reproduce**
This bug is intermittent. However, to reproduce the behavior:
1. Create a question and dashboard; add the question to the dashboard
2. Add a filter to the dashboard
3. Click *Edit dashboard*
4. Click *Edit* (on the filter)
5. Observe that the question is not replaced by a drop-down (see attached)

**Expected behavior**
I expected the question to be replaced by a drop down, indicating the columns to which the filter could be applied.

**Screenshots**

Here is a screenshot of the observed behavior:

![dashboard_filters](https://user-images.githubusercontent.com/477263/101161036-de87d380-3627-11eb-89ca-318f867f9541.png)

Here is a screenshot of the expected behavior:

![dashboard_filters_expected](https://user-images.githubusercontent.com/477263/101161882-73d79780-3629-11eb-9309-7ef4fae482df.png)

To get the second screenshot, I created a new question and added it to the existing dashboard.

**Information about your Metabase Installation:**

Metabase v0.35.4.

```
{
  "browser-info": {
    "language": "en-GB",
    "platform": "Linux x86_64",
    "userAgent": "Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:83.0) Gecko/20100101 Firefox/83.0",
    "vendor": ""
  },
  "system-info": {
    "file.encoding": "UTF-8",
    "java.runtime.name": "OpenJDK Runtime Environment",
    "java.runtime.version": "11.0.7+10",
    "java.vendor": "AdoptOpenJDK",
    "java.vendor.url": "https://adoptopenjdk.net/",
    "java.version": "11.0.7",
    "java.vm.name": "OpenJDK 64-Bit Server VM",
    "java.vm.version": "11.0.7+10",
    "os.name": "Linux",
    "os.version": "4.14.203-156.332.amzn2.x86_64",
    "user.language": "en",
    "user.timezone": "GMT"
  },
  "metabase-info": {
    "databases": [
      "postgres"
    ],
    "hosting-env": "unknown",
    "application-database": "postgres",
    "application-database-details": {
      "database": {
        "name": "PostgreSQL",
        "version": "12.2"
      },
      "jdbc-driver": {
        "name": "PostgreSQL JDBC Driver",
        "version": "42.2.8"
      }
    },
    "run-mode": "prod",
    "version": {
      "date": "2020-05-28",
      "tag": "v0.35.4",
      "branch": "release-0.35.x",
      "hash": "b3080fa"
    },
    "settings": {
      "report-timezone": null
    }
  }
}
```

**Severity**
This will block some of our users this month (December 2020) and next month (January 2021) and could block our usage of Metabase thereafter, unless we could find a work-around.

**Additional context**
None

**Comment by flamber** [comment]
Try upgrading to latest release 0.37.3. I cannot reproduce on latest release.

**Comment by iaindillingham** [comment]
Thanks for looking into this @flamber. I've tested v0.35.4 and v0.37.3 using a new Docker container in each case. I recreated the issue on v0.35.4 by creating a question that referenced a saved question (a native query), adding this question to a dashboard, and then refreshing the database schema. I could not recreate the issue on v0.37.3, so will upgrade our dev/stage/production environments to v0.37.3.
